packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -200,3 +200,113 @@ test('addLogRetentionLifetime modifies the retention period of the custom resour
     RetentionInDays: customResourceLogRetention,
   });
 });
+
+describe('when custom resource logGroup removalPolicy is undefined', () => {
+  test('addRemovalPolicy modifies custom resource logGroup to Delete', () => {
+    // GIVEN
+    const customResourceRemovalPolicy = cdk.RemovalPolicy.DESTROY;
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app);
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 'BucketDeployment', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup', {
+        removalPolicy: cdk.RemovalPolicy.RETAIN, /* Explicitly set to the default value `RETAIN` */
+      }),
+    });
+
+    // WHEN
+    CustomResourceConfig.of(app).addRemovalPolicy(customResourceRemovalPolicy);
+
+    // THEN
+    const template = Template.fromStack(stack);
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResource('AWS::Logs::LogGroup', {
+      UpdateReplacePolicy: 'Delete',
+      DeletionPolicy: 'Delete',
+    });
+  });
+
+  test('addLogRetentionLifetime creates new logGroup and addRemovalPolicy modify the logGroup to removalPolicy Delete', () => {
+    // GIVEN
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const customResourceRemovalPolicy = cdk.RemovalPolicy.DESTROY;
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app);
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 'BucketDeployment', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+
+    // WHEN
+    CustomResourceConfig.of(app).addRemovalPolicy(customResourceRemovalPolicy);
+
+    // THEN
+    const template = Template.fromStack(stack);
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResource('AWS::Logs::LogGroup', {
+      UpdateReplacePolicy: 'Delete',
+      DeletionPolicy: 'Delete',
+    });
+  });
+
+  test('addRemovalPolicy keeps custom resource logGroup to Retain', () => {
```

Comment:
keeps it? As in it isn't modified? I think this test is only interested in asserting that it gets set to `Retain`, so maybe:

```suggestion
  test('addRemovalPolicy can see a custom resource logGroup's removal policy to Retain', () => {
```